registry-scanner/pkg/image/image.go

```diff
@@ -5,20 +5,36 @@ import (
 	"time"
 
 	"github.com/distribution/distribution/v3/reference"
+	"github.com/sirupsen/logrus"
 
 	"github.com/argoproj-labs/argocd-image-updater/registry-scanner/pkg/log"
 	"github.com/argoproj-labs/argocd-image-updater/registry-scanner/pkg/tag"
 )
 
+// ContainerImage represents a container image and its update configuration.
+// It acts as a rich domain object, carrying both its identity and the rules
+// for how it should be updated.
 type ContainerImage struct {
-	RegistryURL           string
-	ImageName             string
-	ImageTag              *tag.ImageTag
-	ImageAlias            string
+	// Identity fields
+	RegistryURL string
+	ImageName   string
+	ImageTag    *tag.ImageTag
+	ImageAlias  string
+
+	// Update settings
+	UpdateStrategy UpdateStrategy
+	ForceUpdate    bool
+	AllowTags      string
+	IgnoreTags     []string
+	PullSecret     string
+	Platforms      []string
+
+	// Manifest-specific fields
 	HelmParamImageName    string
 	HelmParamImageVersion string
 	KustomizeImage        *ContainerImage
-	original              string
+
+	original string
 }
```

Comment:
ContainerImage is part of registry-scanner, which is supposed to be a neutral library. The issue with adding image updater knowledge to this type is that it becomes tied to image updater, and no longer easily reusable outside of image updater.

Reply:
I have created type Image in types.go. 
I moved `GetParameterPullSecret` from registry-scanner to pkg because an original used annotations and this function is not related with registry-scanner.